Program.cs

```diff
@@ -296,13 +296,21 @@ public void Start()
                     // Get one, show some info
                     _parent.Log("Client accepted:" + clientSocket.RemoteEndPoint.ToString());
                     HTTPProcessor hp = new HTTPProcessor(clientSocket, _parent);
-                    hp.Process();
+
+                    Thread t = new Thread(() => foo(hp));
```

Comment:
Don't name a method as "foo", it doesn't mean anything (and it's the same as in your friend's homework  ha ha ha ha ha).